torch/_dynamo/guards.py

```diff
@@ -772,6 +1316,32 @@ def TENSOR_MATCH(self, guard: Guard, value=None):
                 self.tensor_check_examples.append(value)
                 self.tensor_check_guards.append(guard)
 
+                if config.enable_cpp_guard_manager:
+                    guard_manager = self.get_guard_manager(guard)
+                    # Keep track of all the tensor guard managers to insert
+                    # NoAliasing check at the end.
+                    self.tensor_check_guard_managers.append(guard_manager)
+
+                    output_graph = self.check_fn_manager.output_graph
+                    size = convert_to_concrete_values(
```

Comment:
How does this work with dynamic shapes?

Reply:
This is no different from the current TensorGuards. This code is present in `make_check_fn` manager at the end because we used to put TensorGuards at the end earlier. But with CPP, we can put them earlier and install an aliasing guard later.

Maybe the function helper name is bad, when its dynamic shapes, it returns a symint.

Reply:
Yeah, its just that it could be named better. On symint inputs, it can return None.